products/BellHybrid/sys/SystemManager.cpp

```diff
@@ -64,4 +65,35 @@ namespace sys
         auto msg = std::make_shared<CriticalBatteryLevelNotification>(true, charging);
         bus.sendUnicast(std::move(msg), service::name::appmgr);
     }
+
+    void SystemManager::Run()
+    {
+        initialize();
+
+        // in shutdown we need to wait till event manager tells us that it's ok to stfu
+        while (state == State::Running) {
+            if (auto msg = mailbox.pop(); msg) {
+                msg->Execute(this);
+            }
+        }
+
+        while (state == State::Shutdown) {
+            set(State::ShutdownReady);
+        }
+
+        DestroySystemService(service::name::evt_manager, this);
+        CloseService();
+
+        // it should be called before systemDeinit to make sure this log is dumped to the file
+        LogPowerOffReason();
+
+        if (systemDeinit) {
+            systemDeinit();
+        }
+
+        EndScheduler();
+
+        // Power off system
+        PowerOff();
+    }
```

Comment:
What is the purpose?

Reply:
@tomaszrybarski-digica ?

Reply:
IMHO, this is completely wrong (for Pure too) :/

Reply:
This behavior must be separated since Bell needs a ShutdownReady state issued immediately not depending on whether device is in Charging mode or not (USB cable connected / disconnected). The device is being shutdown properly then both when USB is connected or disconnected.
The current implementation of SystemManager::Run() which Bell uses is common for Pure and Bell and in this implementation ShutdownReady is issued only in Discharge mode (USB disconnected). This causes issues in proper shutdown of Bell when device is being charged

Reply:
This also has been tested and device is in proper shutdown mode (both for USB connected and disconnected - charging and discharging states). Logs can be provided

Reply:
The proper way of doing it will be not to override whole `Run()` method but to create some function with pre-shutdown routine with empty implementation for Bell.

Reply:
You mean empty implementation for Pure? The current implementation is correct for Pure. A pre-shutdown function with an implementation of just setting ShutdownReady state could be added for Bell then

Reply:
Actually I meant something like this. in SystemManagerCommon::Run()
`while (state == State::Shutdown) {
  shutdownState(); // pure-virtual in sysMgrCommon
}
`
for Bell:
`shutdownState(){
 set(State::ShutdownReady);
}`

for Pure:
`shutdownState(){
 // Pure things
}`